File: punchy-backend/src/lib/passwordPolicy.ts
import { z } from 'zod';

export const strongPassword = z.string()
  .min(8)
  .regex(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[^A-Za-z0-9]).{8,}$/, 'Password must be at least 8 characters and include an uppercase letter, lowercase letter, number, and symbol.');

File: punchy_app/lib/core/validation/password_policy.dart
class PasswordPolicy {
  const PasswordPolicy._();

  static final RegExp pattern = RegExp(
    r'^(?=.*[a-z])(?=.*[A-Z])(?=.*\d).{8,}$',
  );

  static bool isValid(String value) => pattern.hasMatch(value);

  static const message =
      'Use 8+ characters with uppercase, lowercase and number';
}
